carp.deployment.core/src/commonMain/kotlin/dk/cachet/carp/deployment/domain/DeviceDeploymentStatus.kt

```diff
@@ -9,31 +9,68 @@ import kotlinx.serialization.Serializable
  * Describes the status of a device, part of a study deployment.
  */
 @Serializable
-data class DeviceDeploymentStatus(
+sealed class DeviceDeploymentStatus
+{
     /**
      * The description of the device.
      */
     @Serializable( DeviceDescriptorSerializer::class )
-    val device: AnyDeviceDescriptor,
+    abstract val device: AnyDeviceDescriptor
     /**
      * Determines whether registering the device is required for the deployment to start running.
      */
-    val requiresRegistration: Boolean,
-    /**
-     * Determines whether the device is currently registered.
-     */
-    val isRegistered: Boolean,
+    abstract val requiresRegistration: Boolean
     /**
      * Determines whether the device requires a device deployment by retrieving [MasterDeviceDeployment].
      * Not all master devices necessarily need deployment; chained master devices do not.
      */
-    val requiresDeployment: Boolean,
+    abstract val requiresDeployment: Boolean
+
+
+    /**
+     * A device deployment status which has not been deployed yet.
+     */
+    interface NotDeployed
+    {
+        /**
+         * Determines whether the device has been registered successfully and is ready for deployment.
+         */
+        val isReadyForDeployment: Boolean
+    }
+
+
+    /**
+     * Device deployment status for when a device has not been registered.
+     */
+    @Serializable
+    data class Unregistered(
+        @Serializable( DeviceDescriptorSerializer::class )
+        override val device: AnyDeviceDescriptor,
+        override val requiresRegistration: Boolean,
+        override val requiresDeployment: Boolean,
+        override val isReadyForDeployment: Boolean
+    ) : DeviceDeploymentStatus(), NotDeployed
+
     /**
-     * Determines whether the device has been registered successfully and is ready for deployment.
+     * Device deployment status for when a device has been registered.
      */
-    val isReadyForDeployment: Boolean,
+    @Serializable
+    data class Registered(
+        @Serializable( DeviceDescriptorSerializer::class )
+        override val device: AnyDeviceDescriptor,
+        override val requiresRegistration: Boolean,
+        override val requiresDeployment: Boolean,
+        override val isReadyForDeployment: Boolean
+    ) : DeviceDeploymentStatus(), NotDeployed
+
     /**
-     * True if the device has retrieved its [MasterDeviceDeployment] and was able to load all the necessary plugins to execute the study.
+     * Device deployment status when the device has retrieved its [MasterDeviceDeployment] and was able to load all the necessary plugins to execute the study.
      */
-    val isDeployed: Boolean
-)
+    @Serializable
+    data class Deployed(
+        @Serializable( DeviceDescriptorSerializer::class )
+        override val device: AnyDeviceDescriptor,
+        override val requiresRegistration: Boolean,
+        override val requiresDeployment: Boolean
```

Comment:
Yes. The user might still want to know after a study has started, whether a device originally needed to be registered or not. Otherwise, that information is lost once the state changes.